### DISCUSSION TITLE
[Dev Tools] How can I  install forge zksync without replacing the `forge` command?

### DISCUSSION THREAD

**Discussion by oveddan** [zkSync-Community-Hub/zksync-developers#761] [Dev Tooling] [answered]
### Environment

Mainnet

### Select the Dev Tool you are using

Foundry

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

i want to install the [foundry zksync](https://docs.zksync.io/build/tooling/foundry/installation) extension, but do not want to replace my existing forge command, as I've pinned it to a version that works well. 
Is there a way to install foundry zksync so that i can run something like forge-zk and it runs the zksync version of forge, without replacing my existing forge command?

**Comment by alhenaai** [comment]
To install the Foundry ZKsync extension without replacing your existing `forge` command, you can use an alias or a separate installation directory. Here’s how you can do it:

1. **Install Foundry ZKsync** using the standard installation script, but specify a different installation directory to avoid overwriting your current `forge` installation. You can do this by modifying the installation path in the script or by setting an environment variable if the script supports it.

2. **Create an alias** for the ZKsync version of `forge`. After installation, you can add an alias to your shell configuration file (like `.bashrc` or `.zshrc`). For example:
   ```bash
   alias forge-zk='/path/to/zksync/forge'
   ```
   Replace `/path/to/zksync/forge` with the actual path where the ZKsync version of `forge` is installed.

This setup allows you to use `forge-zk` for ZKsync-specific operations while keeping your original `forge` command unaffected.

For more detailed instructions and options, you can refer to the [Foundry installation guide](https://docs.zksync.io/build/tooling/foundry/installation).

Feel free to visit the [ZKsync developers discussions on Github](https://github.com/zkSync-Community-Hub/zksync-developers/discussions) for more in-depth answers or further assistance.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by dutterbutter** [comment, marked as answer]
@oveddan we are in the process of discussing this proposal with upstream foundry to remedy this issue: https://github.com/foundry-rs/foundry/issues/9227

In the meantime, you have two options:

1. **Compile from Source:** Run `cargo build --bin forge --release` to build the project. Then, use `./target/release/forge` directly, or rename and move `forge` to `/usr/local/bin` for easier access.

2. **Adjust `foundryup-zksync` Scripts:** Update the scripts in [`foundryup-zksync`](https://github.com/matter-labs/foundry-zksync/tree/main/foundryup-zksync) to use a custom directory (instead of the default `.foundry`). Afterward, manually rename the `forge` and `cast` binaries as needed.

**Reply by oveddan** [reply]
Thanks this pointed me in the right direction!

I ended up doing #2.

**Comment by zhanxin-xu** [comment]
I changed the installation script:
```bash
curl -L https://raw.githubusercontent.com/1329576606/foundry-zksync/refs/heads/main/install-foundry-zksync | bash
```
After the command is successfully executed, restart the terminal.
Two new commands will be added :forge-zksync and cast-zksync.

**Reply by oveddan** [reply]
wow thanks, this is a great solution.

**Comment by asselstine** [comment]
The script from @1329576606 was out of date, so I ended up moving .foundry to .foundry-backup, running the install script, then moving it back
